minimalkv/fsspecstore.py

```diff
@@ -95,8 +95,11 @@ def __init__(self, prefix: str = "", mkdir_prefix: bool = True):
             If True, the prefix will be created if it does not exist.
             Analogous to the create_if_missing parameter in AzureBlockBlobStore or GoogleCloudStore.
         """
-        self.prefix = prefix
+        if write_kwargs is None:
+            write_kwargs = {}
+        self._prefix = prefix
```

Comment:
any reason you turned this from public to private? this is technically a breaking change, which is why I'm asking.

Reply:
The `Boto3Store` has a `prefix` attribute, and the two don't have the same meaning. Renaming it to smth like `path_prefix` would also be an option.

Reply:
Ah, shoot. so we had a `prefix` argument in `FSSpecStore` and `Boto3Store`? We'll have to rename at least one (hence breaking the API here). We can do this once we're moving to a `v2.0` release, but not now.

What about this:

- In `FSSPEC` store we rename `prefix` to `_prefix` and add a `prefix` property that triggers a deprecation warning when called (probably we should do this for all instance attributes that shouldn't be touched by the user).
- Instead of changing `Boto3Store`, we add a new store, `S3FSSpecStore`. Calling the `S3FSSpecStore` should trigger a deprecation warning that this store will be renamed to `Boto3Store` in the next major release.

Alternatively, we can also wait with this PR until we have a `v2.0` release.

Reply:
That makes sense.

- I reverted the `Boto3Store`. I deprecated its `prefix` attribute and added `object_prefix`, which will be the replacement
- I added the `S3FSStore`, which doesn't have a `prefix` attribute but only an `object_prefix`. I added a warning that it will be renamed to `Boto3Store`

Reply:
Nice, thanks a lot! What do you think about turning all class attributes in `FSSpecStore` private? Ork maybe the entire class should be private/abstract?

Reply:
Turning all attributes private sounds good. I would keep the `fs` public, and the `FSSpecStore` itself too. That way, any `fsspec` `FileSystem` can be turned into a `KeyValueStore`.